atomic_reactor/plugins/prepub_flatpak_create_oci.py

```diff
@@ -247,6 +254,108 @@ def _export_filesystem(self):
             self.log.info("Cleaning up docker container")
             self.tasker.d.remove_container(container_id)
 
+    def _get_components(self, manifest):
+        with open(manifest, 'r') as f:
+            lines = f.readlines()
+
+        return parse_rpm_output(lines)
+
+    def _check_runtime_manifest(self, components):
+        # For a runtime, we want to make sure that the set of RPMs that was installed
+        # into the filesystem is *exactly* the set that is listed in the 'runtime'
+        # profile. Requiring the full listed set of RPMs to be listed makes it
+        # easier to catch unintentional changes in the package list that might break
+        # applications depending on the runtime. It also simplifies the checking we
+        # do for application flatpaks, since we can simply look at the runtime
+        # modulemd to find out what packages are present in the runtime.
+
+        base_module = self.source.compose.base_module
+
+        component_names = {c['name'] for c in components}
+        expected_component_names = set(base_module.mmd.profiles['runtime'].rpms)
+
+        if component_names != expected_component_names:
+            missing = expected_component_names - component_names
+            extra = component_names - expected_component_names
+            raise RuntimeError("Installed set of packages does not match runtime profile:\n"
+                               "\tmissing: {}\n\textra: {}"
+                               .format(" ".join(sorted(missing)),
+                                       " ".join(sorted(extra))))
+
+        return components
+
+    def _identify_app_source_modules(self):
+        modules = self.source.compose.modules
+        base_module = self.source.compose.base_module
+
+        # Identify the module for the Flatpak runtime that this app runs against
+        runtime_module = None
+        for key in base_module.mmd.buildrequires.keys():
+            try:
+                module = modules[key]
+                if 'runtime' in module.mmd.profiles:
+                    runtime_module = module
```

Comment:
Should we `break` here?

Reply:
Really, really precise handling would be to check that only one of the build dependencies is a runtime module, but I'm not expecting people to make that mistake in practice, so adding another error path seems unnecessary. First is  more self-explanatory than last, so I'll add the break.